libflash/fileio.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef FILEIO_HPP
+#define FILEIO_HPP
+#include "common/io.hpp"
+#include "platformfs.h"
+
+namespace mender {
+namespace io {
+
+class FileWriter : public common::io::Writer {
+public:
+	FileWriter(File f);
```

Comment:
In general we should do as little manual resource management as possible (such as calling `free` and `close`), and I would recommend that `FileWriter` takes ownership of `f`. That means it needs a destructor where you call `Close`, so that it happens automatically.

Together with `GetFile()` it should be appropriate to track the ownership this way I think.